graph/include/LoggerUtil.h

```diff
@@ -32,11 +34,108 @@ class MCGLogger {
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->info(formattedMessage);
+    }
+  };
+
+  template <Output outPutType=Output::DEFAULT, LogType lt = LogType::DEFAULT, typename MSG_t, typename... Args>
+  void error(const MSG_t msg, Args&&... args) {
+    const std::string& formattedMessage = fmt::format(msg, std::forward<Args>(args)...);
+    if (uniqueify<lt>(formattedMessage)){
+      return ;
+    }
+    if constexpr (outPutType==Output::DEFAULT){
+      getConsole()->error(formattedMessage);
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->error(formattedMessage);
+    }
+  };
+
+  template <Output outPutType=Output::DEFAULT, LogType lt = LogType::DEFAULT, typename MSG_t, typename... Args>
+  void debug(const MSG_t msg, Args&&... args) {
+    const std::string& formattedMessage = fmt::format(msg, std::forward<Args>(args)...);
+    if (uniqueify<lt>(formattedMessage)){
+      return ;
+    }
+    if constexpr (outPutType==Output::DEFAULT){
+      getConsole()->debug(formattedMessage);
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->debug(formattedMessage);
+    }
+  };
+
+  template <Output outPutType=Output::DEFAULT, LogType lt = LogType::DEFAULT, typename MSG_t, typename... Args>
+  void warn(const MSG_t msg, Args&&... args) {
+    const std::string& formattedMessage = fmt::format(msg, std::forward<Args>(args)...);
+    if (uniqueify<lt>(formattedMessage)){
+      return ;
+    }
+    if constexpr (outPutType==Output::DEFAULT){
+      getConsole()->warn(formattedMessage);
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->warn(formattedMessage);
+    }
+  };
+
+  template <Output outPutType=Output::DEFAULT, LogType lt = LogType::DEFAULT, typename MSG_t, typename... Args>
+  void critical(const MSG_t msg, Args&&... args) {
+    const std::string& formattedMessage = fmt::format(msg, std::forward<Args>(args)...);
+    if (uniqueify<lt>(formattedMessage)){
+      return ;
+    }
+    if constexpr (outPutType==Output::DEFAULT){
+      getConsole()->critical(formattedMessage);
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->critical(formattedMessage);
+    }
+  };
+
+  template <Output outPutType=Output::DEFAULT, LogType lt = LogType::DEFAULT, typename MSG_t, typename... Args>
+  void trace(const MSG_t msg, Args&&... args) {
+    const std::string& formattedMessage = fmt::format(msg, std::forward<Args>(args)...);
+    if (uniqueify<lt>(formattedMessage)){
+      return ;
+    }
+    if constexpr (outPutType==Output::DEFAULT){
+      getConsole()->trace(formattedMessage);
+    }else if constexpr (outPutType==Output::ERROR){
+      getErrConsole()->trace(formattedMessage);
+    }
+  };
+
  private:
   MCGLogger() : console(spdlog::stdout_color_mt("console")), errconsole(spdlog::stderr_color_mt("errconsole")) {}
 
+
+  template <LogType lt>
+  inline bool uniqueify(const std::string& formattedMessage) {
```

Comment:
```suggestion
  inline bool ensureUnique(const std::string& formattedMessage) {
```
IMHO the word `uniqueify` reads a bit weird.

Reply:
Changed function name to suggested